rswag-specs/lib/rswag/specs/request_factory.rb

```diff
@@ -178,7 +178,7 @@ def build_query_string_part(param, value, swagger_doc)
         end
 
         type = param[:type] || param.dig(:schema, :type)
-        return "#{name}=#{value}" unless type&.to_sym == :array
+        return "#{CGI.escape(name.to_s)}=#{CGI.escape(value.to_s)}" unless type&.to_sym == :array
```

Comment:
This'll work as-is but it looks like there's a lot of repetition for `CGI.escape(name.to_s)` further up in the method.
Might be worth a touch of refactoring with `escaped_name = CGI.escape(name.to_s)`.